### ISSUE TITLE
Lex bot plan and apply fails if nlu_confidence_threshold is set to anything other than 0.5

### ISSUE THREAD

**Issue by awsjasono** [hashicorp/terraform-provider-awscc#1061]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

### Terraform CLI and Terraform AWS Cloud Control Provider Version

Terraform v1.5.3
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v5.6.2
+ provider registry.terraform.io/hashicorp/awscc v0.55.0

### Affected Resource(s)

* awscc_lex_bot

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "awscc_lex_bot" "demo_lexv2_bot" {
  name = "botTest"
  role_arn = aws_iam_service_linked_role.lex_service_role.arn
  data_privacy = {
    child_directed = false
  }
  idle_session_ttl_in_seconds = 300
  auto_build_bot_locales      = true
  bot_locales = [
    {
      locale_id = "en_US",
      nlu_confidence_threshold = 0.6,
      voice_settings = {
        voice_id = "Joanna"
      },
      intents = [
        {
          name = "Sales",
          sample_utterances = [
            {
              utterance = "sales"
            }
          ],
          dialog_code_hook = {
            enabled = false
          },
          fulfillment_code_hook = {
            enabled = false,
            post_fulfillment_status_specification = {
              success_response = {
                message_groups_list = [
                  {
                    message = {
                      plain_text_message = {
                        value = "Transferring call to Sales"
                      }
                    }
                  }
                ],
                allow_interrupt = true
              },
              failure_response = {
                message_groups_list = [
                  {
                    message = {
                      plain_text_message = {
                        value = "error"
                      }
                    }
                  }
                ],
                allow_interrupt = true
              }
            }
          }
        },
        {
          name = "Finance",
          sampleUtterances = [
            {
              utterance = "finance"
            }
          ],
          dialogCodeHook = {
            enabled = false
          },
          fulfillment_codeHook = {
            enabled = false,
            post_fulfillment_status_specification = {
              success_response = {
                message_groups_list = [
                  {
                    message = {
                      plain_text_message = {
                        value = "Transferring call to Finance"
                      }
                    }
                  }
                ],
                allow_interrupt = true
              },
              failure_response = {
                message_groups_list = [
                  {
                    message = {
                      plain_text_message = {
                        value = "error"
                      }
                    }
                  }
                ],
                allow_interrupt = true
              }
            }
          }
        },
        {
          name        = "FallbackIntent",
          description = "Default intent when no other intent matches",
          dialog_code_hook = {
            enabled = false
          },
          fulfillment_code_hook = {
            enabled = false
          },
          parent_intent_signature = "AMAZON.FallbackIntent"
        }
      ]
    }
  ]
}
```

### Expected Behavior

lex bot should be able to successfully run a plan and be deployed using an nlu confidence threshold other than .5

### Actual Behavior

terraform plan fails with

Error: Missing Configuration for Required Attribute

Must set a configuration value for the bot_locales[...trimmed for brevity...].nlu_confidence_threshold
│ attribute as the provider has marked it as required.
│ 
│ Refer to the provider documentation or contact the provider developers for additional information about configurable attributes that are required.

### Steps to Reproduce

1. `terraform plan`
2. 'terraform apply`

### References

* #0000

**Comment by wellsiau-aws** [comment]
only works with `nlu_confidence_threshold` = 0.5

**Comment by wellsiau-aws** [comment]
Smaller configuration to replicate

```
resource "awscc_lex_bot" "demo_lexv2_bot" {
  name = "botTest"
  role_arn = "arn:aws:iam::123456789012:role/aws-service-role/lex.amazonaws.com/AWSServiceRoleForLexBots"
  data_privacy = {
    child_directed = false
  }
  idle_session_ttl_in_seconds = 300
  auto_build_bot_locales      = true
  bot_locales = [
    {
      locale_id = "en_US",
      nlu_confidence_threshold = 0.6
    }
  ]
}
```

To replicate, simply run `terraform validate`
```
│ Error: Missing Configuration for Required Attribute
│ 
│   with awscc_lex_bot.demo_lexv2_bot,
│   on main.tf line 9, in resource "awscc_lex_bot" "demo_lexv2_bot":
│    9:   bot_locales = [
│   10:     {
│   11:       locale_id = "en_US",
│   12:       nlu_confidence_threshold = 0.6
│   13:     }
│   14:   ]
│ 
│ Must set a configuration value for the
│ bot_locales[Value({"custom_vocabulary":<null>,"description":<null>,"intents":<null>,"locale_id":"en_US","nlu_confidence_threshold":0.600000,"slot_types":<null>,"voice_settings":<null>})].nlu_confidence_threshold
│ attribute as the provider has marked it as required.
│ 
│ Refer to the provider documentation or contact the provider developers for additional information about configurable attributes that are required.
```

**Comment by kadrach** [comment]
I cannot pinpoint exactly *why*, but this is fixed by https://github.com/hashicorp/terraform-plugin-framework/pull/817. 

Will be resolved by https://github.com/hashicorp/terraform-provider-awscc/pull/1117.

**Comment by wellsiau-aws** [comment]
confirmed that this issue resolved on awscc v0.60.0
